sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonIndexer.java

```diff
@@ -64,6 +67,22 @@ ProjectLevelSymbolTable projectLevelSymbolTable() {
     return projectLevelSymbolTable;
   }
 
+  void removeFileFromProjectSymbolTable(InputFile inputFile) {
+    String packageName = packageNames.get(inputFile.uri());
+    packageNames.remove(inputFile.uri());
+    PythonFile pythonFile = SonarQubePythonFile.create(inputFile);
+    projectLevelSymbolTable.removeModule(packageName, pythonFile);
+  }
+
+  void addFileToProjectSymbolTable(InputFile inputFile) throws IOException {
```

Comment:
I think that `PythonIndexer` could eventually be used for incremental analysis in general, so not only restricted to ProjectLevelSymbolTable. I would suggest to rename those method names to be less specific, like "addFile" and "removeFile". Wdyt?